model-optimizer/extensions/middle/LayoutChangeForConstantShapePaths.py

```diff
@@ -108,8 +101,7 @@ def find_and_replace_pattern(self, graph: Graph):
             shape.out_port(0).get_connection().insert_node(gather)
 
         # 2. Inserting Gather/Transpose to NC* format
-        shape_sub_graph_end_points = self.find_shape_subgraph_endpoints(
-            [shape.out_port(0) for shape in shape_ops], None, self.mark_node_as_in_correct_layout_by_in_port)
```

Comment:
Could you elaborate on why the `mark_node_as_in_correct_layout_by_in_port` call has been removed?

Reply:
Because `MarkSubGraphsWithCorrectLayout` was extended to mark weights AND shape sub-graphs as in correct layout. A kind of decomposition. So this code is used for shape sub-graph layout change only.